[Bug] grass.script: KeyValue should raise AttributeError when attempting to access a key as an attribute

## Describe the bug

The KeyValue class returns a KeyError instead of an AttributeError when an non-existing attribute is accessed.
One consequence is that it is impossible to do a deepcopy of the object

## To reproduce

```python
import pytest

def test_KeyValue():
    """Check that KeyValue class works as expected"""
    kv = gs.KeyValue()
    kv["key1"] = "value1"
    kv["key2"] = "value2"
    
    assert kv["key1"] == "value1"
    assert kv.key1 == "value1"
    assert kv["key2"] == "value2"
    assert kv.key2 == "value2"
    
    # Check that it raises KeyError for non-existing keys
    with pytest.raises(KeyError):
        _ = kv["non_existing_key"]

    # Check that it raises AttributeError for non-existing attribute
    with pytest.raises(AttributeError):
        _ = kv.non_existing_key
```
```

self = {'key1': 'value1', 'key2': 'value2'}, key = 'non_existing_key'

    def __getattr__(self, key: str) -> VT:
>       return self[key]
               ^^^^^^^^^
E       KeyError: 'non_existing_key'

```

As for the deepcopy issue:

```python
from copy import deepcopy
kv_copy = deepcopy(gs.KeyValue())
```

```
self = {}, key = '__deepcopy__'

    def __getattr__(self, key: str) -> VT:
>       return self[key]
               ^^^^^^^^^
E       KeyError: '__deepcopy__'

```

## Expected behavior
When an attribute is non existent, an AttributeError should be raised

## System description

- Operating System: Ubuntu 25.04
- GRASS GIS version: 110ff3393c

--- a/python/grass/script/tests/utils_test.py
+++ b/python/grass/script/tests/utils_test.py
@@ -25,26 +25,29 @@ def test_unrecognized_separator():
     assert gs.separator("apple") == "apple"
 
 
-def test_KeyValue():
-    """Check that KeyValue class works as expected"""
+def test_KeyValue_keys():
+    """Check that KeyValue class works like a Dict"""
     kv = gs.KeyValue()
     kv["key1"] = "value1"
     kv["key2"] = "value2"
 
     assert kv["key1"] == "value1"
-    assert kv.key1 == "value1"
     assert kv["key2"] == "value2"
-    assert kv.key2 == "value2"
 
-    # Check that it raises KeyError for non-existing keys
+    # Raises KeyError for non-existing keys
     with pytest.raises(KeyError):
         _ = kv["non_existing_key"]
 
-    # Check that it raises AttributeError for non-existing attribute
+
+def test_KeyValue_values():
+    """Check that keys of KeyValue class can be accessed as attributes"""
+    kv = gs.KeyValue()
+    kv["key1"] = "value1"
+    kv["key2"] = "value2"
+
+    assert kv.key1 == "value1"
+    assert kv.key2 == "value2"
+
+    # Raises AttributeError for non-existing attribute
     with pytest.raises(AttributeError):
-        _ = kv.non_existing_key
-
-    # Deep copy works correctly
-    from copy import deepcopy
-
-    _ = deepcopy(gs.KeyValue())
+        _ = kv.non_existing_attribute

--- a/python/grass/script/tests/utils_test.py
+++ b/python/grass/script/tests/utils_test.py
@@ -1,5 +1,7 @@
 """Test functions in grass.script.utils"""
 
+import pytest
+
 import grass.script as gs
 
 
@@ -21,3 +23,28 @@ def test_backslash_separators():
 def test_unrecognized_separator():
     """Check that unknown strings are just passed through"""
     assert gs.separator("apple") == "apple"
+
+
+def test_KeyValue():
+    """Check that KeyValue class works as expected"""
+    kv = gs.KeyValue()
+    kv["key1"] = "value1"
+    kv["key2"] = "value2"
+
+    assert kv["key1"] == "value1"
+    assert kv.key1 == "value1"
+    assert kv["key2"] == "value2"
+    assert kv.key2 == "value2"
+
+    # Check that it raises KeyError for non-existing keys
+    with pytest.raises(KeyError):
+        _ = kv["non_existing_key"]
+
+    # Check that it raises AttributeError for non-existing attribute
+    with pytest.raises(AttributeError):
+        _ = kv.non_existing_key
+
+    # Deep copy works correctly
+    from copy import deepcopy
+
+    _ = deepcopy(gs.KeyValue())

--- a/python/grass/script/utils.py
+++ b/python/grass/script/utils.py
@@ -171,7 +171,10 @@ class KeyValue(dict[str, VT]):
     """
 
     def __getattr__(self, key: str) -> VT:
-        return self[key]
+        try:
+            return self[key]
+        except KeyError:
+            raise AttributeError(key)
 
     def __setattr__(self, key: str, value: VT) -> None:
         self[key] = value

